queries/julia/highlights.scm

```diff
@@ -1,106 +1,113 @@
-(identifier) @variable
+;;; Identifiers
 
-(operator) @operator
-(range_expression ":" @operator)
-(pair_expression "=>" @operator)
+(identifier) @variable
 
-;; In case you want type highlighting based on Julia naming conventions (this might collide with mathematical notation)
-;((identifier) @type ; exception: mark `A_foo` sort of identifiers as variables
-  ;(match? @type "^[A-Z][^_]"))
+; ;; If you want type highlighting based on Julia naming conventions (this might collide with mathematical notation)
+; ((identifier) @type
+;   (match? @type "^[A-Z][^_]"))  ; exception: Highlight `A_foo` sort of identifiers as variables
 
 (macro_identifier) @function.macro
-(macro_identifier (identifier) @function.macro) ; for any one using the variable highlight
+(macro_identifier
+  (identifier) @function.macro) ; for any one using the variable highlight
+
 (macro_definition
-  name: (identifier) @function.macro
-  ["macro" "end" @keyword])
+  name: (identifier) @function.macro)
+
+(quote_expression ":" (identifier)) @symbol
+
+
+;;; Fields and indexes
 
 (field_expression
-  (identifier)
   (identifier) @field .)
 
+(subscript_expression
+  (_)
+  (range_expression
+    (identifier) @constant.builtin .)
+  (#eq? @constant.builtin "end"))
```

Comment:
`end` is not really a constant. It has no value and is rather a keyword. If the range expression is similar to a loop it could be `@repeat`

Reply:
When indexing an array with a range, `end` can be used as a shorthand for the last index. The capture that was being used before was `@number`, and I updated it to `@constant.builtin` in https://github.com/nvim-treesitter/nvim-treesitter/pull/2977

I don't mind the capture that's used here, as long as it isn't `@keyword` because I think the `end` keyword that's used to close blocks is conceptually different from the index `end`.

Reply:
If this is specific to the `end` in `foo[1:end]` and doesn't match `do...end` (please confirm), I -- and likely @theHamsta -- fully agree.

Reply:
> If this is specific to the end in foo[1:end] and doesn't match do...end

Yep. This one is just for indices.

I made a separate rule for `do … end` in line 173 to highlight it as `@function.keyword` since it's just another way to define a lambda.